Wp 207 fix pzp startup on android

aldafu opened the pull request:
If the sdcard isn't available on Android the PZP will fail to start.
- Fix console.log implementation from the PZP that is only used on Android and assumes that the /sdcard path is always available.
- Fix context manager that fails to check if a directory exists before trying to create it.

http://jira.webinos.org/browse/WP-207

aldafu committed:
fix recursive directory creation

check the return value from stat call and only create dir if it doesn't
exist already.

Jira-issue: WP-207

aldafu committed:
Fix writing to console.log on Android

Check if /sdcard path is available before trying to write to it.

Jira-issue: WP-207

aldafu committed:
Update .gitignore

To include generated files form the Android projects.

Jira-issue: WP-207

ghost commented:
Why is there so much changed in the .gitignore file? A lot of files that were ignored are no longer ignored...

aldafu commented:
Actually I only moved stuff around a bit, trying to sort it more logically. And added the Android bits.

I've removed nothing, so more is getting ignored now, if I'm not mistaken.

ghost commented:
I'm sorry, that's right. Merging
